resolwe_bio/processes/alignment/star.yml

```diff
@@ -467,20 +473,39 @@
       {% endif %}
       re-progress 0.8
 
-      # If alignment job produces an empty .bam file, the file needs to be re-sorted
-      # for it to be successfully indexed by samtools.
-      if [[ $(samtools view -c Aligned.sortedByCoord.out.bam) == 0 ]]
-      then
-        samtools sort Aligned.sortedByCoord.out.bam > Aligned.sortedByCoord_temp.out.bam
-        re-checkrc "Samtools sort command failed."
-        samtools index Aligned.sortedByCoord_temp.out.bam "${MATE1_NAME}.bam.bai"
-        re-checkrc "Samtools index command failed."
-        mv Aligned.sortedByCoord_temp.out.bam "${MATE1_NAME}.bam"
-      else
-        samtools index Aligned.sortedByCoord.out.bam "${MATE1_NAME}.bam.bai"
-        re-checkrc "Samtools index command failed."
-        mv Aligned.sortedByCoord.out.bam "${MATE1_NAME}.bam"
-      fi
+      {% if star_sort %}
```

Comment:
`if [[ $(samtools view -c Aligned.sortedByCoord.out.bam) == 0 ]]` has been introduced due to a bug in STAR aligner. This has since been fixed. If we don't need this if/else anymore, perhaps we could simplify it to something along the lines of

```
OUT_NAME = "${MATE1_NAME}.bam"
OUT_NAME_BAI = "${MATE1_NAME}.bai"
OUT_NAME_TEMP = "${OUT_NAME}_temp"

{% if star_sort %}
    IN_NAME = "Aligned.sortedByCoord.out.bam"
{% else %}
    IN_NAME = "Aligned.out.bam"
{% endif %}

samtools sort "${IN_NAME}" > "${OUT_NAME_TEMP}
re-checkrc "Samtools sort command failed."
samtools index "${OUT_NAME_TEMP}" "${OUT_NAME_BAI}"
re-checkrc "Samtools index command failed."
mv "${OUT_NAME_TEMP}" "${OUT_NAME}"
```